libuavcan/include/uavcan/build_config.hpp

```diff
@@ -243,8 +253,7 @@ struct UAVCAN_EXPORT IsDynamicallyAllocatable
 {
     static void check()
     {
-        char dummy[(sizeof(T) <= MemPoolBlockSize) ? 1 : -1] = { '0' };
-        (void)dummy;
+        static_assert((sizeof(T) <= MemPoolBlockSize), "Type cannot be allocated dynamically.");
```

Comment:
So we are officially leaving C++03 behind?

Reply:
Yes.

Reply:
Remember you merged feature/issue100 already. So the switch has been thrown. It's just a matter of cleaning up now and fully embracing the new language. As I stated before: we should not add more c++11 into this PR and instead add it in one feature at a time when I have a nuttx verification environment setup. So far we are using, and should continue to use, static_assert and unique_ptr (i.e. do not use auto_ptr any more).

To that end: what should I do to be able to verify nuttx compatibility with these changes?

Reply:
> what should I do to be able to verify nuttx compatibility with these changes?

Let me invoke @davids5. David, how difficult would it be to set up a testing environment with NuttX on some kind of commonly available STM32H7 demo board like NUCLEO-H743ZI? @LorenzMeier can we get some assistance with that, please?

Reply:
Plain NuttX or PX4/NuttX?

Reply:
Plain NuttX preferred, if possible.